src/javascript/app/Stores/Modules/Contract/contract-store.js

```diff
@@ -126,6 +129,20 @@ export default class ContractStore extends BaseStore {
         }
     }
 
+    handleSubscribeProposalOpenContract = (contract_id, cb) => {
+        // TODO: remove .toString() when API is ready
+        const proposal_open_contract_request = [contract_id.toString(), cb, false];
+
+        if (this.should_forget_first) {
+            WS.forgetAll('proposal_open_contract').then(() => {
+                this.should_forget_first = false;
+                WS.subscribeProposalOpenContract(...proposal_open_contract_request);
```

Comment:
can't we send a should_forget_first param as true to this instead of handling forget ourselves?

Reply:
Subscribing doesn't return a promise, so we can't set `should_forget_first` false after the forgetting is done